catroid/src/main/java/org/catrobat/catroid/stage/PreStageActivity.java

```diff
@@ -563,6 +595,31 @@ private void nfcInitialize() {
 		resourceInitialized();
 	}
 
+	private static ArrayList<Brick> getBricksRequieringResource(int resource) {
```

Comment:
The method name is misleading. All other implementations with the same method name return **all** bricks whereas this also filters duplicate classes. As mentioned a few lines below, `Project.getBricksRequiringResource` will return all bricks. So you just need to create a filter method passing in this list.

I would love to see a generic filter version :wink:

Btw: It's [requ**i**ring](http://www.dictionary.com/browse/requiring)